File: solutions/feb_2024/169_majority_element/solution.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        cur_num = nums[0]
        count = 1

        for num in nums[1:]:
            if num == cur_num:
                count += 1
            else:
                count -= 1
                if count < 0:
                    cur_num = num
                    count = 1

        return cur_num

File: solutions/feb_2024/169_majority_element/solution.java
class Solution {
  public int majorityElement(int[] nums) {
    int curNum = nums[0];
    int count = 1;

    for (int i = 1; i < nums.length; i++) {
      if (nums[i] == curNum) {
        count++;
        continue;
      }

      if (count == 0) {
        curNum = nums[i];
        count = 1;
      } else {
        count--;
      }
    }

    return curNum;
  }
}
